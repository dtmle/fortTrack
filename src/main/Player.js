import React from "react";

export class Player extends React.Component {
  render() {
    return (
      <div className="Player">
        <h2 className="playerName">{this.props.name}</h2>
        {this.props.children}
      </div>
    );
  }
}

export class Stats extends React.Component {
  constructor(props) {
    super(props);
    this.state = { stats: this.props.stats };
  }

  getStatsForMode() {
    const mode = this.props.mode;
    switch (mode) {
      case "Total":
        return this.state.stats.lifeTimeStats;
      case "Solos":
        return this.state.stats.stats.solo;
      case "Duos":
        return this.state.stats.stats.duo;
      case "Squads":
        return this.state.stats.stats.squad;
      default:
        return 0;
    }
  }

  render() {
    const stats = this.getStatsForMode();
    const exists = stats !== undefined ? true : false;
    return (
      <div className="Player">
        <h2>{this.props.name}</h2>
        <div className="StatContainer">
          <p id="matchesPlayed">
            # Matches: <span>{exists ? stats.matches : 0}</span>
          </p>
          <p id="wins">
            Wins: <span>{exists ? stats.wins : 0}</span>
          </p>
          <p id="winratio">
            Win Ratio: <span>{exists ? stats.winRatio + "%" : "0%"}</span>
          </p>
          <p id="kills">
            Kills: <span>{exists ? stats.kills : 0}</span>
          </p>
          <p id="kd">
            K/D: <span>{exists ? stats.kd : 0}</span>
          </p>
        </div>
      </div>
    );
  }
}
