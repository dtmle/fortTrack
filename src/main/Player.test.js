import React from "react";
import { mount, render, shallow } from "enzyme";
import { Player, Stats } from "./Player";

describe("Player", () => {
  it("should render", () => {
    const component = shallow(<Player />);
    expect(component).toMatchSnapshot();
  });
  it("should render name", () => {
    const component = shallow(<Player name={"Dennis"} />);
    expect(component.find("h2").text()).toEqual("Dennis");
  });
  it("should render stats if valid & searched", () => {
    const component = render(<Player searched={true} valid={true} />);
    expect(component.find("p").length).toEqual(4);
  });
});

describe("Stats", () => {
  it("should render", () => {
    const component = shallow(<Stats />);
    expect(component).toMatchSnapshot();
  });
  it("should render name", () => {
    const component = shallow(<Stats name={"Dennis"} />);
    expect(component.find("h2").text()).toEqual("Dennis");
  });
  it("should have 4 p's", () => {
    const component = shallow(<Stats name={"Dennis"} />);
    expect(component.find("p").length).toEqual(4);
  });
  it("should render wins", () => {
    const component = shallow(<Stats wins={100} />);
    expect(component.find("#wins").get(0).props.children[1].props.children).toEqual(100);
  });
  it("should render win percent", () => {
    const component = shallow(<Stats winPercent={100} />);
    expect(component.find("#winratio").get(0).props.children[1].props.children).toEqual(100);
  });
  it("should render kills", () => {
    const component = shallow(<Stats kills={100} />);
    expect(component.find("#kills").get(0).props.children[1].props.children).toEqual(100);
  });
  it("should render k/d", () => {
    const component = shallow(<Stats kd={100} />);
    expect(component.find("#kd").get(0).props.children[1].props.children).toEqual(100);
  });
});
