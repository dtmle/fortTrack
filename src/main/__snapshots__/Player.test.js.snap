// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player should render 1`] = `
<div
  className="Player"
>
  <h2
    className="playerName"
  />
</div>
`;

exports[`Stats should render 1`] = `
<div
  className="Player"
>
  <h2 />
  <div
    className="StatContainer"
  >
    <p
      id="wins"
    >
      Wins: 
      <span />
    </p>
    <p
      id="winratio"
    >
      Win Ratio: 
      <span />
    </p>
    <p
      id="kills"
    >
      Kills: 
      <span />
    </p>
    <p
      id="kd"
    >
      K/D: 
      <span />
    </p>
  </div>
</div>
`;
